scenes/cslib_menu.txt

```diff
@@ -105,38 +107,54 @@
 *label build_simple_with_cancel
 *params
 *temp allow_cancel true
-*goto _build_simple_common
+*gosub _build_simple_common
+*gosub _render_choice
+*return
+
+
+*comment ::: INTERNAL FUNCTIONS BELOW (DO NOT CALL DIRECTLY!) :::
+*label _build_simple_common
+*gosub _init_state
+*temp prefix "param"
+*temp filter ""
+*temp filter_value ""
+*temp page_count "@{(param_count <= const_items_per_page) 1|${(param_count / const_items_per_page)}}"
+*if ((page_count modulo 1) > 0)
+	*set page_count ((page_count - (page_count modulo 1)) + 1)
+*gosub _populate_page
+*return
 
 
-*comment INTERNAL FUNCTIONS (DO NOT CALL DIRECTLY):
-*label _common
+*label _build_array_common
 *params p_prefix p_cancel p_filter p_filter_type p_filter_value
+*gosub _init_state
 *temp prefix p_prefix
 *temp filter p_filter
 *temp filter_value p_filter_value
 *temp filter_type p_filter_type
 *if (filter != "") and ((filter_type != "MATCH") and (filter_type != "NOT_MATCH"))
 	*bug filter_type was "${filter_type}" but should be "MATCH" or "NOT_MATCH"
 *temp allow_cancel p_cancel
+*gosub _reset_filter_and_populate_page
 *return
 
-*label _build_simple_common
-*temp prefix "param"
-*temp filter ""
-*temp filter_value ""
 
-*label _init
-*comment CONSTANT CONFIG VALUES:
-*comment swap the next/prev buttons intelligently for smoother navigation
-*temp next_preference true
+*label _init_state
+*comment determines the next/prev option position for smoother navigation
```

Comment:
Again, this is very ambitious. A version with no pagination would be easier to understand and debug. BTW, do you really expect a menu with multiple pages of choices?

Reply:
See my other comment somewhere above: I basically think that removing pagination would make this module redundant. Maybe you need to actually play with this patch yourself? Because I actually found it surprisingly easy to test pagination. It's quite fun actually.

You don't need to use the test framework, you can just fire up a normal choice, works a charm.

```
    *comment configure automatic choice selection
    *set cslib_catch_choice true
    *comment dictate which options (comma separated, for nested choices) should be selected
    *set cslib_choice_select "3"
    *fake_choice
      # Go east
        Eastward bound.
      # Go west
        Westward bound.
      # Go south
        Southward bound.
    *comment 'Southward bound.' will be printed here.
    *comment And helper variables will be populated:
    ${cslib_choice_count} == 3
    ${cslib_choice_1} == "Go east."
    ${cslib_choice_2} == "Go west."
    *comment etc.
    *comment and turn off automatic choice selection
    *set cslib_catch_choice false
```

Reply:
thanks!